src/Summoner/Source.hs

```diff
@@ -49,3 +57,7 @@ fetchSource = \case
 
     fetchUrl :: Text -> IO (Maybe Text)
     fetchUrl url = Just . toText <$> readProcess "curl" [toString url] ""
+
+    putLink :: IO (Maybe Text)
+    putLink = errorMessage "See this document under the [following link](<link goes here>)"
```

Comment:
@ChShersh I don't quite think I have this right or if it's what you meant in your previous comment?

Reply:
@Cmdv my ideas was that for the `Link` constructor there should be no IO actions. So you can just write:

```haskell
putLink link = pure $ Just $ "See full content of the file [here](" <> link <> ")"
```

Or you can even use `text` QuasiQuoter from `neat-interpolation` library that we're using in `summoner`:

```haskell
Link link -> pure $ Just [text|See full content of the file [here](${link})|]
```

Reply:
great thanks I see what you mean now, `QuasiQuoter` is nice for sure. 👍

Reply:
actually one question:
`fetchSource :: Source -> IO (Maybe Text)` needs the IO, or should I change that too? because I've gone with:
``` haskell
putLink :: Text -> IO (Maybe Text)
putLink link = pure $ Just [text|See full content of the file [here]($link)|]
```

Reply:
@Cmdv It's okay to have `pure $ Just`. All right sides of `case ... of` expression should have the same type. So it's okay to add some `pure`s